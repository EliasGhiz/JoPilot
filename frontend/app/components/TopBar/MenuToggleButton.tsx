// MenuToggleButton.tsx â€“ Renders the icon button that toggles the sidebar visibility.

import React from "react";
import MenuIcon from "@mui/icons-material/Menu";
import MenuOpenIcon from "@mui/icons-material/MenuOpen";
import { getThemeColor } from "app/theme/colorSystem";
import type { ThemeVariant } from "app/theme/colorSystem";
import type { PaletteMode } from "@mui/material";
import CentralIconButton from "../IconButton/IconButton";

interface MenuToggleButtonProps {
  open: boolean;
  onClick?: () => void;
  hamburgerWhite: boolean;
  colorMode: PaletteMode;
  themeVariant: ThemeVariant;
}

const MenuToggleButton: React.FC<MenuToggleButtonProps> = ({ 
  open, 
  onClick, 
  colorMode,
  themeVariant
}) => {
  const iconColor = "#fff";
  const hoverBgColor = getThemeColor(
    themeVariant,
    'primary',
    colorMode,
    colorMode === 'dark' ? 70 : 80
  );
  
  return (
    <CentralIconButton
      onClick={onClick}
      hoverBgColor={hoverBgColor}
      iconColor={iconColor}
    >
      {open ? <MenuOpenIcon /> : <MenuIcon />}
    </CentralIconButton>
  );
};

export default MenuToggleButton;
