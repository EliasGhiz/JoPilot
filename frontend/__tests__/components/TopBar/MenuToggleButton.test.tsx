import React from 'react';
import { render } from '@testing-library/react';
import MenuToggleButton from '../../../app/components/TopBar/MenuToggleButton';

// Mock all imports
jest.mock('../../../app/theme/themeColors', () => ({
  getThemeColor: () => '#000',
}));

describe('MenuToggleButton Snapshot Test', () => {
  it('should match the snapshot when open', () => {
    const { asFragment } = render(
      <MenuToggleButton
        open={true}
        onClick={() => {}}
        hamburgerWhite={false}
        colorMode="light"
        themeVariant="blue"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match the snapshot when closed', () => {
    const { asFragment } = render(
      <MenuToggleButton
        open={false}
        onClick={() => {}}
        hamburgerWhite={false}
        colorMode="light"
        themeVariant="blue"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});