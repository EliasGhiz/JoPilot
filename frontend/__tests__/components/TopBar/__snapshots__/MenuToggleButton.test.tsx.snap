// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuToggleButton Snapshot Test should match the snapshot when closed 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1tvuiyo"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1oq5es3-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiBox-root css-jt3tvc"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
          />
        </svg>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`MenuToggleButton Snapshot Test should match the snapshot when open 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1tvuiyo"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1oq5es3-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiBox-root css-jt3tvc"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="MenuOpenIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h13v-2H3zm0-5h10v-2H3zm0-7v2h13V6zm18 9.59L17.42 12 21 8.41 19.59 7l-5 5 5 5z"
          />
        </svg>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
